SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -20,6 +22,17 @@ namespace SonarQube.MSBuild.Tasks
     /// </summary>
     public class WriteProjectInfoFile : Task
     {
+        /// <summary>
+        /// Regular expression to validate setting ids.
+        /// </summary>
+        /// <remarks>
+        /// Validation rules:
+        /// Must start with an alpanumeric character.
+        /// Can be followed by any number of alphanumeric characters or .
+        /// Whitespace is not allowed
+        /// </remarks>
+        private static readonly Regex ValidSettingKeyRegEx = new Regex(@"^\w[\w\d\.-]*$", RegexOptions.Compiled);
```

Comment:
This regular expression is similar but not identical to the reg ex in the task (modified above). There wasn't a particularly clean way of sharing the common part, so I didn't.